packages/components/src/components/file-uploader/file-uploader.js

```diff
@@ -159,15 +182,55 @@ class FileUploader extends mixin(
    * @private
    */
   _handleDeleteButton = evt => {
-    const target = eventMatches(evt, `[data-for=${this.inputId}]`);
+    const target = eventMatches(evt, this.options.selectorCloseButton);
     if (target) {
-      this._changeState('delete-filename-fileuploader', {
+      this.changeState('delete-filename-fileuploader', {
         initialEvt: evt,
-        filenameElement: target.parentNode,
+        filenameElement:
+          target.closest(this.options.selectorSelectedFileInvalidWrapper) ||
```

Comment:
Depending on the reason we use two selectors, we may need to introduce a new selector combining the two. @emyarod Could you share what's behind the "two selectors" approach here?

Reply:
I actually [refactored invalid selected items so we wouldn't need two selectors](https://github.com/carbon-design-system/carbon/pull/3175/commits/84b0bcd0832358d7f80e52a48ee9c69d3dd92ffb), just forgot to remove the old selector here